visualizer/visualizer-server/src/solidity/types.rs

```diff
@@ -35,12 +46,37 @@ fn output_mask(field_mask: Option<FieldMask>) -> Result<OutputMask, anyhow::Erro
     Ok(output_mask)
 }
 
+fn fix_sources_paths(sources: BTreeMap<PathBuf, String>) -> BTreeMap<PathBuf, String> {
+    sources
+        .into_iter()
+        .map(|(path, content)| {
+            let path = path
+                .as_os_str()
+                .to_str()
+                .map(|path_str| {
+                    let path_str = sanitize_path(path_str.trim_start_matches('/'));
+                    Path::new(&path_str).to_path_buf()
+                })
+                .unwrap_or(path);
+            (path, content)
+        })
+        .collect()
+}
+
+// Should be the same as in
+// https://github.com/ethereum/sourcify/blob/d0882a5d6158d0f56e121835d79860034f072cd8/services/verification/src/services/Injector.ts#L907
+fn sanitize_path(path: &str) -> String {
+    let path = REGEX_ONLY_CHARS.replace_all(path, "_");
+    let path = REGEX_NO_DOTS.replace_all(path.as_ref(), "_");
```

Comment:
Nit: what is the reason behind implementing regex strings as LazyStatic?

Reply:
to be sure that they are compiled when the service is started